File: year2020/month12/day1216/ms_1216.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> solution(long long n);

int main() {
	long long n;

	cin >> n;

	if (1 <= n && n <= 10000000000) {
		vector<int> answer = solution(n);

		for (int i = 0; i < answer.size(); i++) {
			cout << answer[i];

			if (i != answer.size()) {
				cout << ' ';
			}
		}
	}

	return 0;
}

vector<int> solution(long long n) {
	vector<int> answer;

	while (n > 0) {
		answer.push_back(n % 10);
		n /= 10;
	}

	return answer;
}

File: year2020/month12/day1216/ms_1216.cs
using System;

namespace algorithm_123
{
    class ms_1216
    {
        static int[] solution(long n) 
        {
            int[] answer = new int[0];
            int i = 0;

            while(n > 0) 
            {
                if(answer.Length == i) 
                {
                    int[] temp = new int[i + 1];

                    for(int j = 0; j < answer.Length; j++)
                        temp[j] = answer[j];
                    
                    answer = temp;
                    continue;
                }

                answer[i] = Convert.ToInt32(n%10);
                n /= 10;
                i++;
            }

            return answer;
        }

        static void Main(string[] args)
        {
            long n = Convert.ToInt32(Console.ReadLine());
            
            if(1 <= n && n <= 10000000000) 
            {
                int[] answer = solution(n);
                
                for(int i = 0; i < answer.Length; i++) 
                {
                    Console.Write(answer[i]);

                    if(i != answer.Length)
                    {
                        Console.Write(' ');
                    }
                }
            }
        }
    }
}
